terraform/variables.tf

```diff
@@ -0,0 +1,42 @@
+variable "app_name" {
+  description = "Application name"
+  type        = string
+}
+
+variable "model_name" {
+  description = "Model name"
+  type        = string
+}
+
+variable "channel" {
+  description = "Charm channel"
+  type        = string
+  default     = "latest/stable"
+}
+
+variable "revision" {
+  description = "Charm revision"
+  type        = number
+  nullable    = true
+  default     = null
+}
+
+variable "config" {
+  description = "Config options as in the ones we pass in juju config"
+  type        = map(string)
+  default     = {}
+}
+
+variable "units" {
+  description = "Number of units"
+  type        = number
+  default     = 1
+}
+
+variable "trust" {
+  description = "Equiv of juju deploy --trust"
+  type        = bool
+  default     = false
+}
```

Comment:
As I said in the spec discussion, I am not sure having this as an input makes sense. In my experience, the charm either requires it or not. Having this as an input would only allow broken configuration in that case. Are you aware of charms that can run without `trust`, but that can also benefit from it?